### PR TITLE
Improve accuracy of progress bar for Health Records updates

### PR DESCRIPTION
**Pull request by U-DON**
Resolves department-of-veterans-affairs/vets.gov-team#9576.

### CODE REVIEW

**Review by dhall — COMMENTED** [review]
This LGTM, but I'll defer to @saneshark for approval

**Review by saneshark — APPROVED** [review]
@U-DON since this is extremely difficult to test on staging could we sync for 5-10 minutes today and walk me through the logic for the reducer before merging? 

I looked it over and LGTM. Just want to double check before merging.

**Pull request merged**
